src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -104,30 +81,54 @@ class JesBackend extends Backend with LazyLogging {
     }
   }
 
+  sealed trait JesParameter {
+    def name: String
+    def gcs: String
+    def local: Path
+    def parameterType: String
+
+    final val isInput = this.isInstanceOf[JesInput]
```

Comment:
I would remove entirely the isInput and isOutput thing, which is only used to `filter` `JesInput`s or `JesOutput`s from a list of `JesParamaters` ( `jesParameters.filter(_.isInput)` ), and use `collect` instead (`jesParameters.collect({ case i: JesInput => i})`).